api/storage_service.cc

```diff
@@ -103,7 +103,7 @@ static bool any_of_keyspaces_use_tablets(const http_context& ctx) {
 
 locator::host_id validate_host_id(const sstring& param) {
```

Comment:
` locator: host_id_or_endpoint: keep value as variant `

TBH I really don't like this commit, I think it's a bit controversial.

Is it actually needed to achieve the goal of this PR?

Reply:
why don't you like it?
why is it controversial?
That said, I might get around it and use the current class as it if it get in the way for pushing this PR through

Reply:
IIRC without this patch there is a problem resolving the ip address of ignored nodes for replace and we get the `Host ID {} not found in the cluster` exception since the hosts are not in token_metadata topology yet.

Reply:
kept in v11 as per the above